src/prefect/core/task.py

```diff
@@ -262,6 +265,8 @@ def __init__(
         else:
             self.result = result
 
+        self.target = target
```

Comment:
We should consider how we intend this to interact with the `result` kwarg and in paritcular `result.location`:
- should we require only one be provided?
- should `target` instead accept a `Result` instance, and then we raise an error if both `result` and `target` are provided?
- will `target` become the `location` of `task.result` ?
- should `target` be a boolean and then we use `result.location` as the filepath?

Reply:
- I personally think that if a target is provided then result is also required
- No I like keeping target a simple string and separate
- It definitely should
- Interesting idea 🤔 if we did that then why even have target and not a kwarg like `cache_if_result_exists`?

Reply:
Yea I think there should be only one obvious way of configuring this, so the fact that `Result` exposes a `location` kwarg makes me think `target` should either be a boolean or a `Result` object.  I think having `target` be a full `Result` object is clearest (with an error if `result` is also provided) but that's just me.  **cc:** @lauralorenz for input here

Reply:
Soo I think that target should be a string and result should be separately required to configure everything but location. I like this independently of anything else because it’s basically syntactic sugar and brings us closest to the Luigi/Make simplicity. Everything else kind of falls out of that for me. 

Tbh thinking about it out of nowhere don’t hate it if a Result wrote to its location AND it’s target if both were configured, and the latter is used for the exists pipeline, if we were feeling greedy about writes. But our Results aren’t designed to hold 2 paths. Then I this this becomes about how we have to decide which to choose if both are configured. We could raise an error but that is probably not convenient until the flow is running, so then instead I end up voting for overwriting location with target. Which I think in conclusion means I agree with @joshmeek 🤓

I imagine it will be possible in this set up that someone reuses the same Result object for each of their tasks and vary the target Leary as an easy syntax to split out location for multiple tasks too, especially if we put cache time semantics on targets like our original plan.

Reply:
@lauralorenz yea the pattern you mention in your last paragraph could easily be achieved by doing `with Flow("blah", result=S3Result(bucket="default")):` and then only providing a target on a per-task basis.  

OK cool so the only unknown is how to handle both a provided location and a provided target.

Reply:
Yeah, I have a feeling it will annoy people if we throw exceptions but it will confuse people if we simply overwrite. The latter with a warning log is probably the safest in our philosophy of convenient defaults.

Reply:
OK cool: @joshmeek could you add a warning if `result.location is not None and target is not None` ?  In that case we should do something like 
```python
task_result = result.copy()
task_result.location = target
self.result = task_result
```
(store target as the new location but only for this task instance, we don't want to accidentally alter an object shared by other tasks)

Reply:
@cicdw Would this happen in task.py when the task is initialized?